icu4c/source/common/uniquecharstr.h

```diff
@@ -48,21 +50,41 @@ class UniqueCharStrings {
 
     /**
      * Adds a string and returns a unique number for it.
-     * The string's buffer contents must not change, nor move around in memory,
-     * while this UniqueCharStrings is in use.
-     * The string contents must be NUL-terminated exactly at s.length().
-     *
+     * If copy is false, then the string's buffer contents must not change,
+     * nor move around in memory, while this UniqueCharStrings is in use.
      * Best used with read-only-alias UnicodeString objects that point to
      * stable storage, such as strings returned by resource bundle functions.
+     * The string contents must be NUL-terminated exactly at s.length().
+     *
+     * If copy is true, a copy of the string will be stored and freed internally.
      */
-    int32_t add(const UnicodeString &s, UErrorCode &errorCode) {
+    int32_t add(const UnicodeString &s, bool copy, UErrorCode &errorCode) {
```

Comment:
Why not just make the function be defined to take a NUL-terminated `const char16_t*` with no length parameter and let the callers sort this out?

Reply:
that make sense.